src/main/java/com/google/devtools/build/lib/sandbox/DockerCommandLineBuilder.java

```diff
@@ -153,10 +164,15 @@ public ImmutableList<String> build() {
         "-v", sandboxExecRoot.getPathString() + ":" + execRootInsideDocker.getPathString());
     dockerCmdLine.add("-w", execRootInsideDocker.getPathString());
 
-    for (ImmutableMap.Entry<String, String> additionalMountPath : additionalMounts) {
+    final Set<String> writeablePaths = writableFilesAndDirectories.stream().map(Path::getPathString).collect(Collectors.toSet());
```

Comment:
```suggestion
    final Set<String> writeablePaths = writableFilesAndDirectories.stream().map(Path::getPathString).collect(ImmutableSet.toImmutableSet());
```
and then import the static method.